term/src/main/jni/termExec.cpp

```diff
@@ -71,8 +70,8 @@ static void android_os_Exec_setPtyUTF8Mode(JNIEnv *env, jobject clazz, jint fd,
         tios.c_iflag &= ~IUTF8;
     }
 
-    if (my_tcsetattr(fd, TCSANOW, &tios) < 0)
```

Comment:
Thanks, I guess I broke this when trying to support api level 4. Tcsetattr
is not available at level 4, so the dll does not load. Do you have any
suggestions as to how to support both api 4 and recent api levels? Maybe
looking up tcsetattr at runtime?
On Mon, Apr 6, 2015 at 4:17 AM Alexander-- <notifications@github.com> wrote:

> In term/src/main/jni/termExec.cpp
> <https://github.com/jackpal/Android-Terminal-Emulator/pull/418#discussion_r27792976>
> :
>
> > @@ -71,8 +70,8 @@ static void android_os_Exec_setPtyUTF8Mode(JNIEnv *env, jobject clazz, jint fd,
> >          tios.c_iflag &= ~IUTF8;
> >      }
> >
> > -    if (my_tcsetattr(fd, TCSANOW, &tios) < 0)
>
> This use of TCSANOW is where issue seems to lie. Compare
> https://android.googlesource.com/platform/bionic/+/master/libc/bionic/termios.cpp
> and
> https://android.googlesource.com/platform/bionic/+/android-4.2_r1/libc/include/termios.h
> (hint: the former does not redefine TCSANOW at all).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/jackpal/Android-Terminal-Emulator/pull/418/files#r27792976>
> .
>